gothemis/compare/compare.go

```diff
@@ -77,9 +77,16 @@ const (
 
 // Errors returned by Secure Comparator.
 var (
-	ErrMissingSecret = errors.NewWithCode(errors.InvalidParameter, "empty secret for Secure Comparator")
-	ErrMissingData   = errors.NewWithCode(errors.InvalidParameter, "empty comparison message for Secure Comparator")
-	ErrOverflow      = errors.NewWithCode(errors.NoMemory, "Secure Comparator cannot allocate enough memory")
+	ErrAppendSecret             = errors.New("Failed to append secret")
+	ErrCreateComparatorObject   = errors.New("Failed to create comparator object")
+	ErrDestroyComparatorObject  = errors.New("Failed to destroy comparator object")
+	ErrGetCmpData               = errors.New("Failed to get compare data")
+	ErrGetCmpResult             = errors.New("Failed to get compare result")
+	ErrGetOutput                = errors.New("Failed to get output")
+	ErrGetOutputSize            = errors.New("Failed to get output size")
```

Comment:
It isn't really obvious what "output" is and how exactly it is different from "compare data". Using "get" here is also not very obvious since the user isn't calling any getter or something, they do either `Begin` or `Proceed`.

I'd suggest to do the following:

| Old error | will become | new error |
| -- | -- | -- |
| `ErrGetCmpData` | replace with | `ErrProtocolData` |
| `ErrGetCmpResult` | rename into | `ErrNoResult` |
| `ErrGetOutput` | rename into | `ErrProtocolData` |
| `ErrGetOutputSize` | rename into | `ErrProtocolDataSize` |

with appropriate updates in error messages.